final-project/helpers.py

```diff
@@ -167,20 +167,20 @@ def populate_categories_in_database():
 
 
 # Find our record for this product
-def find_product(product_requested):
+def find_product(id_requested):
 
     # Create a parameter to search our database
     # for any records, which contain the product name.
-    product_parameter = "%" + product_requested + "%"
+    product_parameter = "%" + id_requested + "%"
 
     #Search the database for the requested product.
     product_data = db.execute("""
         SELECT *
         FROM products
-        WHERE product_name
-        LIKE :product
+        WHERE id
```

Comment:
Since this is an id, this should be a direct lookup, rather than a query.

Reply:
It does not look like this is possible with SQLite.

Here is a conversation about it
http://sqlite.1065341.n5.nabble.com/Direct-access-of-table-data-td24816.html

Here is the closest SQLite documentation I could find.
https://sqlite.org/queryplanner.html

I will keep this in mind for the future if I switch to PostgreSQL

Reply:
I mean, you have a limit 30, which doesn't make much sense.

https://stackoverflow.com/a/11206275/2204868

My SQL is rusty, but I think it would be this:
```python
db.execute("SELECT * FROM products WHERE id = :product_id", product_id=input_id)
```

What I'm saying is that this shouldn't even be plural in the first place.